sources/Object/Paths.ts

```diff
@@ -1,41 +1,83 @@
 import {Key} from '../Any/Key'
-import {NonNullableFlat} from '../Object/NonNullable'
+import {NonNullableFlat} from './NonNullable'
 import {Cast} from '../Any/Cast'
 import {List} from '../List/List'
 import {BuiltIn} from '../Misc/BuiltIn'
 import {Primitive} from '../Misc/Primitive'
 import {Length} from '../List/Length'
 import {Keys} from '../Any/Keys'
 
+/**
+ * @hidden
+ */
+type PathMode = 'compact' | 'all' | 'required'
+
 /**
  * @hidden
  */
 type UnionOf<A> =
     A extends List
     ? A[number]
-    : A[keyof A]
+    : Exclude<A[keyof A], undefined>
```

Comment:
This fixes an edge case (I added tests for it) that causes optional fields (ie `{a?:number}`) to be converted to paths that contain `List<Key>` which in turn case cause `String.Join` to fail and fallback to `string`